File: tool/src/org/antlr/v4/automata/SetTransition.java
package org.antlr.v4.automata;

import org.antlr.v4.misc.IntervalSet;
import org.antlr.v4.tool.Grammar;

/** A transition containing a set of values */
public class SetTransition extends Transition {
	public IntervalSet label;

	public SetTransition(IntervalSet label, NFAState target) {
		super(target);
		if ( label==null ) label = IntervalSet.of(Label.INVALID);
		this.label = label;
	}

	public IntervalSet label() { return label; }

	public int compareTo(Object o) {
		return 0;
	}

	//	public boolean intersect(Label other) {
//		if ( other.getClass() == SetTransition.class ) {
//			return label.and(((SetTransition)other).label).isNil();
//		}
//		return label.member(((AtomTransition)other).label);
//	}
	
	public int hashCode() {	return label.hashCode(); }

	public boolean equals(Object o) {
		if ( o==null ) return false;
		if ( this == o ) return true; // equals if same object
		if ( o.getClass() == AtomTransition.class ) {
			o = IntervalSet.of(((AtomTransition)o).label);
		}
		return this.label.equals(((SetTransition)o).label);
	}

	public String toString(Grammar g) {
		return label.toString(g);
	}

	public String toString() {
		return label.toString();
	}	
}

File: runtime/JavaScript/src/antlr4/transition/SetTransition.js
// A transition containing a set of values.
import IntervalSet from "../misc/IntervalSet.js";
import Token from '../Token.js';
import Transition from "./Transition.js";

export default class SetTransition extends Transition {
    constructor(target, set) {
        super(target);
        this.serializationType = Transition.SET;
        if (set !==undefined && set !==null) {
            this.label = set;
        } else {
            this.label = new IntervalSet();
            this.label.addOne(Token.INVALID_TYPE);
        }
    }

    matches(symbol, minVocabSymbol, maxVocabSymbol) {
        return this.label.contains(symbol);
    }

    toString() {
        return this.label.toString();
    }
}
